src/Microsoft.HttpRepl/HttpState.cs

```diff
@@ -289,5 +289,17 @@ public Uri GetEffectivePath(IReadOnlyList<string> sections, bool requiresBody, o
 
             return builder.Uri;
         }
+
+        internal string GetRelativePathString()
```

Comment:
Why not just make this public?

IMO, methods should be either public (part of the behavior contract of the type), or private (implementation detail).  I prefer to avoid internal methods (the same rules don't apply for types).

Reply:
That's fair. I changed it to public. My instinct tends to be "as restricted as possible to make it still work" but I see your point.